packages/discovery-provider/src/queries/comments/utils.py

```diff
@@ -444,3 +459,140 @@ def build_comments_query(
         )
 
     return query
+
+
+def format_comments(
+    session, comments, current_user_id, include_replies=False, artist_id=None
+):
+    """
+    Format a list of comments, optionally including their replies.
+
+    Args:
+        session: Database session
+        comments: List of comment query results (tuples from the database)
+        current_user_id: ID of the user making the request
+        include_replies: Whether to include replies in the response
+        artist_id: ID of the track owner (for track comments and replies)
+
+    Returns:
+        List of formatted comments
+    """
+    from src.queries.comments.replies import get_replies
+
+    # Pre-fetch all replies if needed
+    replies_map = {}
+    if include_replies:
+        for comment_data in comments:
+            comment = comment_data[0]  # First element is always the comment
+            # For tombstone comments (deleted with replies), we need to fetch replies
+            # even if the comment is deleted
+            replies = get_replies(
```

Comment:
Out of scope for this PR, but the separate query per comment to fetch replies is not my favorite.